libs/utils/src/rotation.test.ts

```diff
@@ -0,0 +1,170 @@
+  ).toBe(2);
+
+  const nonRotationSingleCandidateContest = electionSample.contests.find(
+    (c) => c.type === 'candidate' && c.seats === 1
+  ) as CandidateContest;
+  expect(
+    getContestCandidatesInRotatedOrder({
+      contest: nonRotationSingleCandidateContest,
+      precinctIndex: 0,
+    }).findIndex((c) => c.id === 'barchi-hallaren')
+  ).toBe(0);
+});
+
+test('Get contest vote in rotated order', () => {
+  const multiSeatCandidateContest = electionSampleRotation.contests.find(
+    (c) => c.type === 'candidate' && c.seats > 1 && c.allowWriteIns
+  ) as CandidateContest;
+  expect(
+    getContestVoteInRotatedOrder({
+      contest: multiSeatCandidateContest,
+      precinctIndex: 0,
+      vote: [],
+    })
+  ).toMatchInlineSnapshot(`Array []`);
+  expect(
+    getContestVoteInRotatedOrder({
+      contest: multiSeatCandidateContest,
+      precinctIndex: 0,
+      vote: [
+        {
+          id: '__writeIn',
+          name: 'first write-in',
+          isWriteIn: true,
+        },
+        multiSeatCandidateContest.candidates[1] as Candidate,
+        multiSeatCandidateContest.candidates[2] as Candidate,
+        multiSeatCandidateContest.candidates[0] as Candidate,
+        {
+          id: '__writeIn',
+          name: 'last write-in',
+          isWriteIn: true,
+        },
+      ],
+    }).map((c) => c.name)
+  ).toMatchInlineSnapshot(`
+    Array [
+      "Camille Argent 1",
+      "Chloe Witherspoon-Smithson 2",
+      "Clayton Bainbridge 3",
+      "first write-in",
+      "last write-in",
+    ]
+  `);
+  expect(
+    getContestVoteInRotatedOrder({
+      contest: multiSeatCandidateContest,
+      precinctIndex: 1,
+      vote: [
+        {
+          id: '__writeIn',
+          name: 'first write-in',
+          isWriteIn: true,
+        },
+        multiSeatCandidateContest.candidates[1] as Candidate,
+        multiSeatCandidateContest.candidates[2] as Candidate,
+        multiSeatCandidateContest.candidates[0] as Candidate,
+        {
+          id: '__writeIn',
+          name: 'last write-in',
+          isWriteIn: true,
+        },
+      ],
+    }).map((c) => c.name)
+  ).toMatchInlineSnapshot(`
+    Array [
+      "Chloe Witherspoon-Smithson 2",
+      "Clayton Bainbridge 3",
+      "Camille Argent 1",
+      "first write-in",
+      "last write-in",
+    ]
+  `);
+
+  const nonRotatedMultiSeatCandidateContest = electionSample.contests.find(
+    (c) => c.type === 'candidate' && c.seats > 1 && c.allowWriteIns
+  ) as CandidateContest;
+  expect(
+    getContestVoteInRotatedOrder({
+      contest: nonRotatedMultiSeatCandidateContest,
+      precinctIndex: 1,
+      vote: [
+        {
+          id: '__writeIn',
+          name: 'first write-in',
+          isWriteIn: true,
+        },
+        nonRotatedMultiSeatCandidateContest.candidates[1] as Candidate,
+        nonRotatedMultiSeatCandidateContest.candidates[2] as Candidate,
+        nonRotatedMultiSeatCandidateContest.candidates[0] as Candidate,
```

Comment:
Unclear to me whether these are better as a cast or as a non-null assertion (which are allowed in tests). What do you think?

Reply:
Leaning non-null assertion as it's clear it should be a Candidate from the code.